plugins/modules/grafana_folder.py

```diff
@@ -246,6 +292,82 @@ def delete_folder(self, folder_uid):
         response = self._send_request(url, headers=self.headers, method="DELETE")
         return response
 
+    def get_folder_permission(self, folder_uid):
+        url = "/api/folders/{folder_uid}/permissions".format(folder_uid=folder_uid)
+        response = self._send_request(url, headers=self.headers, method="GET")
+        for items in response:
+            del items['created']
+            del items['inherited']
+            del items['slug']
+            del items['teamAvatarUrl']
+            del items['teamEmail']
+            del items['title']
+            del items['uid']
+            del items['updated']
+            del items['url']
+            del items['userAvatarUrl']
+            del items['userEmail']
+            del items['userLogin']
+            del items['folderId']
+            del items['permissionName']
+            del items['team']
+            del items['isFolder']
+            if items['userId'] == 0:
+                del items['userId']
+            if items['teamId'] == 0:
+                del items['teamId']
+        return response
+
+    def create_folder_permission(self, folder_uid, permissions):
+        url = "/api/folders/{folder_uid}/permissions".format(folder_uid=folder_uid)
+        items = {
+            "items": []
+        }
+        items["items"].extend(permissions)
+        response = self._send_request(url, data=items, headers=self.headers, method="POST")
+        return response
+
+    def get_user_id_from_mail(self, email):
+        url = "/api/users/lookup?loginOrEmail={email}".format(email=email)
+        user = self._send_request(url, headers=self.headers, method="GET")
+        if user is None:
+            self._module.fail_json(failed=True, msg="User '%s' does not exists" % email)
+        return user.get("id")
+
+    def get_team(self, name):
```

Comment:
We already have code in `grafana_team.py` doing this (line 241).

Reply:
yes, that is something I'd like to do at some point ; have all Grafana API call related stuff in module_utils and have our library that we can use in all modules.

I guess it will be easier if you move the team related stuff that you need to have the big change that I will never have time to review properly.

Sorry for the delay on this PR.